src/fe_modes.c

```diff
@@ -12,11 +12,24 @@
  *
  */
 
+// clang-format off
 int (*mode_functions[])(State *, WINDOW *, WINDOW *) = {
     mode_picker,
     mode_insert,
     mode_pan,
     mode_free_line,
+    mode_brush,
+};
+// clang-format on
+
+typedef struct {
+  char pattern;
+  enum { PAINT_ON, PAINT_OFF } state;
+} mode_brush_config_t;
```

Comment:
This needed some more specific state info, so I made another struct for it rather than add on to the main one.
Would this be a good time to add mode-specific structs to the API?

Reply:
Other Me: No, do that in #45.